### PR TITLE
Correct asset variant documentation

### LINKED ISSUE #8452: Update 'Asset variants' on 'Adding assets and images' page
### Page URL

https://docs.flutter.dev/development/ui/assets-and-images/

### Page source

https://github.com/flutter/website/tree/main/src/development/ui/assets-and-images.md

### Describe the problem

[Asset variants](https://docs.flutter.dev/development/ui/assets-and-images#asset-variants) states:

> When an asset’s path is specified in the assets section of pubspec.yaml, the build process looks for any files with the same name in adjacent subdirectories. Such files are then included in the asset bundle along with the specified asset.

As of https://github.com/flutter/flutter/commit/cb5b5c34, this is no longer the case. The flutter tool's automatic addition of files to the asset bundle is now limited to [resolution-aware image assets](https://docs.flutter.dev/development/ui/assets-and-images#resolution-aware). For example, if a project's pubspec.yaml file declares `mountain.png` as an asset, then `2.0x/mountain.png` would be bundled, but `night-time/mountain.png` would not.

The change has understandably caused confusion (for example, https://github.com/flutter/flutter/issues/121474), and this page should be updated.

### Expected fix

Consider updating [Declaring resolution-aware image assets](https://docs.flutter.dev/development/ui/assets-and-images#resolution-aware) to describe the automatic bundling of resolution-aware variants. Then, [Asset variants](https://docs.flutter.dev/development/ui/assets-and-images#asset-variants) can be deleted entirely. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by andrewkolos**
Fixes #8452, updating the [Adding assets and images](https://docs.flutter.dev/ui/assets-and-images) page.

Because the idea of asset variants is limited to resolution-aware image assets, this PR deletes the outdated [Asset variants ](https://docs.flutter.dev/ui/assets-and-images#asset-variants) section and moves the relevant information to the documentation of [resolution-aware image assets](https://docs.flutter.dev/ui/assets-and-images#resolution-aware). 

Other things this PR does:
* Moves information about the resolution-aware image asset feature to be after the more fundamental information about how to load an image.
* Renames the "Declaring resolution-aware image assets" section to "Resolution-aware image assets".
* Adds a subsection to this, called "Bundling of resolution-aware image assets". This is to provide the information that the Asset variants section provided. A subsection was chosen to match the existing scheme of the [Asset images in package dependencies](https://docs.flutter.dev/ui/assets-and-images#from-packages) section.

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/ui/assets-and-images.md [hunk 1]
```diff
@@ -40,10 +40,9 @@ flutter:
 ```
 
 {{site.alert.note}}
-  Only files located directly in the directory are included unless
-  there are files with the same name inside a subdirectory 
-  (see [Asset Variants](#asset-variants)). To add files located in subdirectories, create 
-  an entry per directory.
+ Only files located directly in the directory are included
+  (see [Resolution-aware image assets](#resolution-aware) for
+  an exception). To add files located in subdirectories, create an entry per directory.
```
Comment (atsansone) [inline src/ui/assets-and-images.md:45]: **issue:** Remove parenthetical.
```suggestion
 This only includes files located in the directory.
  To make an exception, check out [Resolution-aware image assets](#resolution-aware).
  To add files located in subdirectories, create one entry per directory.
```

File: src/ui/assets-and-images.md [hunk 2]
```diff
@@ -162,14 +112,26 @@ Future<String> loadAsset() async {
...
 ### Loading images
 
-Flutter can load resolution-appropriate images for
-the current device pixel ratio.
+To load an image, use the [`AssetImage`][]
+class in a widget's `build()` method.
 
-#### Declaring resolution-aware image assets {#resolution-aware}
+For example, your app can load the background
+image from the asset declarations above:
```
Comment (atsansone) [inline src/ui/assets-and-images.md:119]: **issue:** Don't use above per [GSG](https://developers.google.com/style/word-list#above).
```suggestion
For example, your app can load the background
image from the previous asset declarations:
```

File: src/ui/assets-and-images.md [hunk 3]
```diff
@@ -162,14 +112,26 @@ Future<String> loadAsset() async {
...
+```
+
+### Resolution-aware image assets {#resolution-aware}
+
+Flutter can load resolution-appropriate images for
+the current [device pixel ratio][].
+
+Specifically, [`AssetImage`][] understands how to map a logical requested
 asset onto one that most closely matches the current
 [device pixel ratio][].
```
Comment (atsansone) [inline src/ui/assets-and-images.md:133]: **nit:** Remove the adverb. It adds nothing to the understanding. Not sure if `AssetImage` can "understand", but that's a minor point.
```suggestion
[`AssetImage`][] understands how to map a logical requested
asset onto one that most closely matches the current
[device pixel ratio][].
```

File: src/ui/assets-and-images.md [hunk 4]
```diff
@@ -162,14 +112,26 @@ Future<String> loadAsset() async {
...
+
 In order for this mapping to work, assets should be arranged
```
Comment (atsansone) [inline src/ui/assets-and-images.md:135]: **nit:** `In order for` and `For` mean the same thing. Reduce the clutter.
```suggestion
For this mapping to work, assets should be arranged
```

File: src/ui/assets-and-images.md [hunk 5]
```diff
@@ -211,36 +177,27 @@ That is, if `.../my_icon.png` is 72px by 72px, then
 but they both render into 72px by 72px (in logical pixels),
 if width and height are not specified.
 
-Each entry in the asset section of the `pubspec.yaml`
+{{site.alert.note}}
+  [Device pixel ratio][] depends on [MediaQueryData.size][] which requires to have either a [MaterialApp][] or [CupertinoApp][] as an ancestor of your [`AssetImage`][].
```
Comment (atsansone) [inline src/ui/assets-and-images.md:181]: **Issue:** Use [semantic breaks](https://sembr.org/).
```suggestion
  [Device pixel ratio][] depends on [MediaQueryData.size][].
  This requires to have either a [MaterialApp][] or [CupertinoApp][] as an ancestor of your [`AssetImage`][].
```

File: src/ui/assets-and-images.md [hunk 6]
```diff
@@ -211,36 +177,27 @@ That is, if `.../my_icon.png` is 72px by 72px, then
...
+{{site.alert.end}}
+
+#### Bundling of resolution-aware image assets {#resolution-aware-bundling}
+
+Only the main asset (or its parent directory) needs
+to specified in the assets section of `pubspec.yaml`; the
+variants will be bundled automatically. Each entry
```
Comment (atsansone) [inline src/ui/assets-and-images.md:188]: **issue:** Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).
```suggestion
You only need to specify the main asset or its parent directory
in the `assets` section of `pubspec.yaml`.
Flutter bundles the variants for you.
Each entry
```

**Review by atsansone — CHANGES_REQUESTED** [review]
Fix the semantic breaks and shorten some sentences for readability.

File: src/ui/assets-and-images.md (hunk 1 above)
Comment (andrewkolos) [reply src/ui/assets-and-images.md]: Addressed with commit [88b87f1](https://github.com/flutter/website/pull/8710/commits/88b87f1369ecfc3b2b08fe87d4054937df2cc065)

File: src/ui/assets-and-images.md (hunk 2 above)
Comment (andrewkolos) [reply src/ui/assets-and-images.md]: Addressed with commits [fix image asset loading example](https://github.com/flutter/website/pull/8710/commits/92306970835201e0ce8fd2e33c746a1b40616438) and [update BackgroundImage docregion](https://github.com/flutter/website/pull/8710/commits/fae0b3c8a3eca2feb98e60928d7875e30704e639)

File: src/ui/assets-and-images.md (hunk 3 above)
Comment (andrewkolos) [reply src/ui/assets-and-images.md]: Addressed with commit [remove anthropomorphism of class and adverb](https://github.com/flutter/website/pull/8710/commits/754dc833c7eb3a5a5ad1133a49bc47732436eb73)

**Review by atsansone — APPROVED** [review]
LGTM.

**Review by sfshaza2 — APPROVED** [review]
lgtm!  Thanks, @andrewkolos!

**Comment by khanhnwin** [comment]
Thanks @andrewkolos!

**Pull request merged**